components/NewRepoModal.js

```diff
@@ -6,13 +6,22 @@ import {
   StyleSheet,
   TouchableOpacity,
 } from 'react-native';
+import { connect } from 'react-redux';
 import { Actions } from 'react-native-router-flux';
+import { AddItemAction } from '../actions'
 
-export default class NewRepoModal extends Component {
+class NewRepoModal extends Component {
   state = {
     modalText: '',
   };
 
+  gettingApi = async () => {
```

Comment:
`gettingApi` doesn't completely describe this process intention neither the action raised. You can change this method name to `handlePress` and move the request logic to another method called `gettingApi`. So, at the `handlePress` you'll use the `gettingApi` to fetch the information, call the addItem action and then move back to the RepoList screen.